lib/mox.ex

```diff
@@ -211,33 +216,41 @@ defmodule Mox do
               "module #{inspect(behaviour)} is not a behaviour, please pass a behaviour to :for"
 
       true ->
-        :ok
+        behaviour
     end
   end
 
-  defp define_mock_module(name, behaviour) do
-    funs =
-      for {fun, arity} <- behaviour.behaviour_info(:callbacks) do
-        args = 0..arity |> Enum.to_list() |> tl() |> Enum.map(&Macro.var(:"arg#{&1}", Elixir))
+  defp generate_mock_funs(behaviour) do
+    for {fun, arity} <- behaviour.behaviour_info(:callbacks) do
+      args = 0..arity |> Enum.to_list() |> tl() |> Enum.map(&Macro.var(:"arg#{&1}", Elixir))
 
-        quote do
-          def unquote(fun)(unquote_splicing(args)) do
-            Mox.__dispatch__(__MODULE__, unquote(fun), unquote(arity), unquote(args))
-          end
+      quote do
+        def unquote(fun)(unquote_splicing(args)) do
+          Mox.__dispatch__(__MODULE__, unquote(fun), unquote(arity), unquote(args))
         end
       end
+    end
+  end
 
+  defp define_mock_module(name, mock_funs) do
     info =
-      quote do
-        # Establish a compile time dependency between the mock and the behaviour
-        _ = unquote(behaviour).module_info(:module)
+      for {behaviour, _} <- mock_funs do
+        quote do
+        # Establish a compile time dependency between the mock and the behaviours
+          _ = unquote(behaviour).module_info(:module)
 
-        def __mock_for__ do
-          unquote(behaviour)
+          def __mock_for__ do
+            unquote(behaviour)
```

Comment:
You are defining multiple clauses of the same function. Maybe we should just have it always return a list of behaviours?